### PR TITLE
Upgrade PuppetJs, Polymer and update CHANGELOG

### PR DESCRIPTION
**Pull request by alshakero**
I would appreciate your advice on the apprpiate heading for the CHANGELOG.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,21 @@
...
+- Upgraded Polymer to 1.8.1 [#4146](https://github.com/Starcounter/Starcounter/issues/4146).
+- Upgraded PuppetJs to 2.4.1 [#4147](https://github.com/Starcounter/Starcounter/issues/4147).
+- Upgraded `fast-json-patch` to [1.1.8](https://github.com/Starcounter-Jack/JSON-Patch/releases).
+- Upgraded `json-patch-queue` to [[2.0.1](https://github.com/Palindrom/json-patch-queue/releases).
+- Upgraded `juicy-composition` to [0.1.0](https://github.com/Juicy/juicy-composition/releases).
+- Upgraded `puppet-redirect` [0.4.4](https://github.com/Palindrom/palindrom-redirect/releases}.
+
+## [Unreleased] - Unreleased
+
+### Added
```
Comment (warpech) [inline CHANGELOG.md]: In Changelog, please add the items to the existing `[Unreleased] - Unreleased` section, not start a new one.

File: bower-list.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+Starcounter C:\Work\Starcounter
```
Comment (warpech) [inline bower-list.txt]: Pls remove this file. We already maintain a `bower-list.txt` file here: https://github.com/Starcounter/Starcounter/blob/develop/src/BuildSystem/ClientFiles/bower-list.txt

You need to run `bower list > bower-list.txt` from the `src/BuildSystem/ClientFiles` directory.

File: src/BuildSystem/ClientFiles/StaticFiles/sys/puppet-redirect/.bower.json [hunk 3]
```diff
@@ -1,9 +1,9 @@
 {
-  "name": "puppet-redirect",
-  "version": "0.4.3",
-  "description": "Custom Element that redirects to a new URL when an attribute is changed. It's also being listened by PuppetJS.",
+  "name": "palindrom-redirect",
+  "version": "0.4.4",
+  "description": "Custom Element that redirects to a new URL when an attribute is changed. It's also being listened by Palindrom.",
```
Comment (warpech) [inline src/BuildSystem/ClientFiles/StaticFiles/sys/puppet-redirect/.bower.json]: Let's not introduce palindrom in this PR. Please revert to puppet-redirect 0.4.3

**Comment by warpech** [comment]
Implements #4146, #4147

File: CHANGELOG.md [hunk 4]
```diff
@@ -61,6 +47,12 @@
 - Improved number of accesses to bound properties in TypedJSON when the whole viewmodel is requested [#4111](https://github.com/Starcounter/Starcounter/issues/4111)
 
 ### Changed
+- Upgraded Polymer to 1.8.1 [#4146](https://github.com/Starcounter/Starcounter/issues/4146).
+- Upgraded PuppetJs to 2.4.1 [#4147](https://github.com/Starcounter/Starcounter/issues/4147).
+- Upgraded `fast-json-patch` to [1.1.8](https://github.com/Starcounter-Jack/JSON-Patch/releases).
+- Upgraded `json-patch-queue` to [[2.0.1](https://github.com/Palindrom/json-patch-queue/releases).
+- Upgraded `juicy-composition` to [0.1.0](https://github.com/Juicy/juicy-composition/releases).
+- Upgraded `puppet-redirect` to [0.4.3](https://github.com/Palindrom/palindrom-redirect/releases}.
 - Changed the parser for Json-by-example from F# parser to NewtonSoft (nuget) [#3811](https://github.com/Starcounter/Starcounter/issues/3811)
```
Comment (warpech) [inline CHANGELOG.md]: Close, close.... But in this "Changed" section there is already mention of Polymer 1.8.0 and other client side libraries, so please remove redundant information and cluster client-side related entries together.

Comment (alshakero) [inline CHANGELOG.md]: I was hesitant if I should keep both. Will clean it in a second.

**Comment by alshakero** [comment]
@warpech BTW, 

1- v2.3 is 443 commits behind develop. 
2- Someone is commiting to v2.3 still, [one hour ago](https://github.com/Starcounter/Starcounter/commits/v2.3).

Should I merge still?

**Comment by alemoi** [comment]
> Should I merge still?

Yes, regardless of other people commits one should merge. Though, when there is a merging conflict one should contact the person created that conflict and forgot to merge it to develop.

**Pull request merged**